src/right-field/results-table/ResultsTable.jsx

```diff
@@ -0,0 +1,129 @@
+import "./ResultsTable.css"
+
+import config from "../../config.json"
+import { useEffect, useState } from "react";
+import { Grid } from 'gridjs-react';
+import "gridjs/dist/theme/mermaid.min.css";
+config = JSON.parse(JSON.stringify(config))
+
+if(!config.queryFolder){
+  config.queryFolder = './'
+}
+
+if(config.queryFolder.substring(config.queryFolder.length-1) !== '/'){
+  config.queryFolder = `${config.queryFolder}/`
+}
+
+const QueryEngine = require('@comunica/query-sparql').QueryEngine;
+const myEngine = new QueryEngine()
+
+/**
+ * 
+ * @param {query} props.query The query (as defined in the config file) that should be executed and results displayed in the table. 
+ * @returns {Component} A React component giving a structural representation of the query results.  
+ */
+function ResultsTable(props){
+    const selectedquery = props.selectedquery
+    const [results, setResults] = useState([])
+    const [variables, setVariables] = useState([])
+    let adder = (item) => setResults((old) => {return [...old, item]})
+    let onqueryChanged = () => {
+      if(selectedquery){
+        setResults([])
+        executequery(selectedquery, adder, setVariables)
+      }
+    }
+
+    if(props.refreshButton.current){
+      props.refreshButton.current.onclick = onqueryChanged
+    }
+
+    useEffect(() => {
+        onqueryChanged()
+    }, [selectedquery])
+
+    return(
+        <div className="results-table">
+            {!selectedquery && <label>Please select a query.</label>}
+            {selectedquery && 
+            <Grid style={{td: {"text-align": "center"}, th: {"text-align": "center"}}} 
+            data={results} 
+            fixedHeader={true}
+            height={"100%"}
+            autoWidth="true" columns={variables}/>
+            }
+        </div>
+    )
+}
+
+/**
+ * A function that executes a given query and processes every result as a stream based on the functions provided. 
+ * @param {query} query the query which gets executed 
+ * @param {Function} adder a function which handles what happens with every variable result  
+ * @param {Function} variableSetter a function which handles what happens with every variable name 
+*/
+function executequery(query, adder, variableSetter){
+    return fetch(`${config.queryFolder}${query.queryLocation}`).then(result => {
```

Comment:
I would use async/await instead of promises.

Reply:
I have adjusted this in [commit 327a8c8](https://github.com/SolidLabResearch/generic-data-viewer/pull/1/commits/327a8c8e1503c5c8fbca4ea08eb0a61d2b52e0d6)